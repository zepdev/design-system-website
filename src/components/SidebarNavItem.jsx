import React, { Component } from 'react'
import PropTypes from 'prop-types'
import Button from './Button'
import { Link } from 'gatsby'

class SidebarNavItem extends Component {
  render() {
    const { item } = this.props

    return (
      <li>
        <Button fullWidth>{item.title}</Button>
        {item['sub-nav'] && (
          <ul className="zds-sidebar--list">
            {Object.keys(item['sub-nav']).map((elem, idx) => (
              <li key={`sni-${idx}`}>
                <Link
                  to={`/content/${item.title
                    .toLowerCase()
                    .replace(/ /g, '-')}/${item['sub-nav'][elem].title
                    .toLowerCase()
                    .replace(/ /g, '-')}`}
                  className="zds-sidebar--nav-subitem zds-button zds-button__full"
                >
                  {item['sub-nav'][elem].title}
                </Link>
              </li>
            ))}
          </ul>
        )}
      </li>
    )
  }
}

SidebarNavItem.propTypes = {
  item: PropTypes.object.isRequired,
}

export default SidebarNavItem
