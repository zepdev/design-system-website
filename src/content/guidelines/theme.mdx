---
label: Guidelines
title: Theme
---

# Theme

Themes are used to customize existing elements to fit the specific style of a strategic business unit or product. There is no base color for the Zeppelin-element-library, almost all elements require that a theme class is present. At the moment the Zeppelin Design System offers three main themes. Developers can customize all elements by adding the theme class to any html tag (`<main>`,`<body>`,`<div>`, ...) as long it wraps all elements you want to use.

```js
<main class="theme-zeppelin" id="app-root">
  ...
</main>
```

We currently offer three main themes;

```js
// Zeppelin Indigo Blue
<div class="theme-zeppelin" >...</div>

// CAT Yellow
<div class="theme-cat" >...</div>

// Rental Red
<div class="theme-rental" >...</div>
```
