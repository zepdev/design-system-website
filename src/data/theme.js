module.exports = {
  breakpoints: {
    xs: '36em' /* 576px */,
    s: '48em' /* 768px */,
    m: '62em' /* 992px */,
    l: '75em' /* 1200px */,
    xl: '105em' /* 1680px */,
    xxl: '120em' /* 1920px */,
  },
  typography: {
    typefaces: {
      roboto: "'Roboto', Arial, Helvetica, sans-serif",
      teko: "Teko', sans-serif",
    },
    basics: {
      fontSizeBase: {
        specification: 'font-size-base',
        px: '16px',
        rem: '1rem',
        weight: null,
      },
      lineHeightBase: {
        specification: 'line-height-base',
        px: '24px',
        rem: '1.5rem',
        weight: null,
      },
      fontWeightBase: {
        specification: 'font-weight-base',
        px: null,
        rem: null,
        weight: 400,
      },
    },
    headings: {
      normal: {
        'zep-typo--normal-1': {
          rem: '0.75rem',
          px: '12px',
          weight: 400,
          specifications: [
            'font-family: "Roboto"',
            'font-weight: 400',
            'line-height: 1.5',
            'text-transform: "none"',
          ],
        },
        'zep-typo--normal-2': {
          rem: '0.75rem',
          px: '12px',
          weight: 400,
          specifications: [
            'font-family: "Roboto"',
            'font-weight: 400',
            'line-height: 1.5',
            'text-transform: "none"',
          ],
        },
        'zep-typo--normal-3': {
          rem: '0.75rem',
          px: '12px',
          weight: 400,
          specifications: [
            'font-family: "Roboto"',
            'font-weight: 400',
            'line-height: 1.5',
            'text-transform: "none"',
          ],
        },
        'zep-typo--normal-4': {
          rem: '0.75rem',
          px: '12px',
          weight: 400,
          specifications: [
            'font-family: "Roboto"',
            'font-weight: 400',
            'line-height: 1.5',
            'text-transform: "none"',
          ],
        },
        'zep-typo--normal-5': {
          rem: '0.75rem',
          px: '12px',
          weight: 400,
          specifications: [
            'font-family: "Roboto"',
            'font-weight: 400',
            'line-height: 1.5',
            'text-transform: "none"',
          ],
        },
        'zep-typo--normal-6': {
          rem: '0.75rem',
          px: '12px',
          weight: 400,
          specifications: [
            'font-family: "Roboto"',
            'font-weight: 400',
            'line-height: 1.5',
            'text-transform: "none"',
          ],
        },
        'zep-typo--normal-7': {
          rem: '0.75rem',
          px: '12px',
          weight: 400,
          specifications: [
            'font-family: "Roboto"',
            'font-weight: 400',
            'line-height: 1.5',
            'text-transform: "none"',
          ],
        },
        'zep-typo--normal-8': {
          rem: '0.75rem',
          px: '12px',
          weight: 400,
          specifications: [
            'font-family: "Roboto"',
            'font-weight: 400',
            'line-height: 1.5',
            'text-transform: "none"',
          ],
        },
      },
      display: {
        'zep-typo--display-1': {
          rem: '0.75rem',
          px: '12px',
          weight: 400,
          specifications: [
            'font-family: "Roboto"',
            'font-weight: 400',
            'line-height: 1.5',
            'text-transform: "none"',
          ],
        },
        'zep-typo--display-2': {
          rem: '0.75rem',
          px: '12px',
          weight: 400,
          specifications: [
            'font-family: "Roboto"',
            'font-weight: 400',
            'line-height: 1.5',
            'text-transform: "none"',
          ],
        },
        'zep-typo--display-3': {
          rem: '0.75rem',
          px: '12px',
          weight: 400,
          specifications: [
            'font-family: "Roboto"',
            'font-weight: 400',
            'line-height: 1.5',
            'text-transform: "none"',
          ],
        },
        'zep-typo--display-4': {
          rem: '0.75rem',
          px: '12px',
          weight: 400,
          specifications: [
            'font-family: "Roboto"',
            'font-weight: 400',
            'line-height: 1.5',
            'text-transform: "none"',
          ],
        },
      },
      special: {
        'zep-typo--special-1': {
          rem: '0.75rem',
          px: '12px',
          weight: 400,
          specifications: [
            'font-family: "Roboto"',
            'font-weight: 400',
            'line-height: 1.5',
            'text-transform: "none"',
          ],
        },
        'zep-typo--special-2': {
          rem: '0.75rem',
          px: '12px',
          weight: 400,
          specifications: [
            'font-family: "Roboto"',
            'font-weight: 400',
            'line-height: 1.5',
            'text-transform: "none"',
          ],
        },
        'zep-typo--special-3': {
          rem: '0.75rem',
          px: '12px',
          weight: 400,
          specifications: [
            'font-family: "Roboto"',
            'font-weight: 400',
            'line-height: 1.5',
            'text-transform: "none"',
          ],
        },
        'zep-typo--special-4': {
          rem: '0.75rem',
          px: '12px',
          weight: 400,
          specifications: [
            'font-family: "Roboto"',
            'font-weight: 400',
            'line-height: 1.5',
            'text-transform: "none"',
          ],
        },
      },
    },
  },
  colors: {
    primary: {
      indigoBlue: {
        name: 'indigo-blue',
        hex: '#27166f',
        rgb: '39/22/111',
        hsb: null,
      },
      catYellow: {
        name: 'cat-yellow',
        hex: '#f0b823',
        rgb: '240/184/35',
        hsb: null,
      },
      rentalRed: {
        name: 'rental-red',
        hex: '#af0e0e',
        rgb: '175/14/14',
        hsb: null,
      },
    },
    gray: {
      grayDark: {
        name: 'gray-dark',
        hex: '#101111',
        rgb: '16/17/17',
        hsb: null,
      },
      grayMidDark: {
        name: 'gray-mid-dark',
        hex: '#373d3f',
        rgb: '55/61/63',
        hsb: null,
      },
      gray: { name: 'gray', hex: '#464a4c', rgb: '70/74/76', hsb: null },
      grayLight: {
        name: 'gray-light',
        hex: '#6f7c80',
        rgb: '111/124/128',
        hsb: null,
      },
      grayLighter: {
        name: 'gray-lighter',
        hex: '#eceeef',
        rgb: '236/238/239',
        hsb: null,
      },
      grayMid: {
        name: 'gray-mid',
        hex: '#f2f3f4',
        rgb: '242/243/244',
        hsb: null,
      },
      grayLightest: {
        name: 'gray-lightest',
        hex: '#f7f7f9',
        rgb: '247/247/249',
        hsb: null,
      },
      white: { name: 'white', hex: '#ffffff', rgb: '255/255/255', hsb: null },
    },
    support: {
      alert: { name: 'alert', hex: '#ff6562', rgb: '255/101/98', hsb: null },
      warning: {
        name: 'warning',
        hex: '#f0ad4e',
        rgb: '240,/73/78',
        hsb: null,
      },
      success: { name: 'success', hex: '#5cb85c', rgb: '92/184/92', hsb: null },
      info: { name: 'info', hex: '#5bc0de', rgb: '91/192/222', hsb: null },
    },
  },
  spacing: {
    xxs: 4,
    xs: 8,
    s: 12,
    m: 16,
    l: 24,
    xl: 32,
    xxl: 48,
  },
}
