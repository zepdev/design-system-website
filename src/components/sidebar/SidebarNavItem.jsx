import React, { useState } from 'react'
import PropTypes from 'prop-types'
import { Link } from 'gatsby'
import clsx from 'clsx'
import withStyles from 'react-jss'
import ButtonBase from '../button/ButtonBase'
import ChevronDownIcon from '../icons/ChevronDownIcon'
import theme from 'zeppelin-element-library/bundle/themes/theme.json'

const styles = theme => ({
  button: {
    textTransform: 'capitalize',
    display: 'flex',
    justifyContent: 'space-between',
    alignItems: 'center',
    padding: `${theme.spacing.component.s.px}px ${theme.spacing.component.xl.px}px`,
    '&:hover': {
      background: theme.colors.gray.grayLightest.hex,
    },
  },
  buttonActive: {
    background: theme.colors.gray.grayLightest.hex,
  },
  list: {
    listStyle: 'none',
    padding: 0,
    margin: 0,
  },
  subnav: {
    textTransform: 'capitalize',
    padding: `${theme.spacing.component.s.px}px ${theme.spacing.component.xl
      .px * 1.5}px`,
    '&:hover': {
      background: theme.colors.gray.grayLightest.hex,
    },
  },
  icon: {
    color: theme.colors.gray.grayLight.hex,
  },
  iconActive: {
    WebkitTransform: 'rotate(180deg)',
    MsTransform: 'rotate(180deg)',
    transform: 'rotate(180deg)',
    transition: 'all 0.75s 0.25s',
  },
  iconReset: {
    WebkitTransform: 'rotate(0deg)',
    MsTransform: 'rotate(0deg)',
    transform: 'rotate(0deg)',
    transition: 'all 0.75s 0.25s',
  },
})

function SidebarNavItem({ item, setMenu, home, link, classes }) {
  const [isButtonOpen, setButton] = useState(false)
  return (
    <li>
      {link && (
        <a
          href={link}
          target="_blank"
          alt={item.title}
          rel="noopener norefferer"
          className={clsx(
            classes.button,
            'zep-typo--normal-body1',
            'zep-button--full'
          )}
        >
          {item.title}
        </a>
      )}
      {!link && item.subnav && (
        <ButtonBase
          fullWidth
          onClick={() => setButton(!isButtonOpen)}
          className={clsx(classes.button, 'zep-typo--normal-body1', {
            [classes.buttonActive]: isButtonOpen,
          })}
        >
          {item.title}
          <ChevronDownIcon
            width="20"
            height="20"
            className={clsx(classes.icon, {
              [classes.iconActive]: isButtonOpen,
              [classes.iconReset]: !isButtonOpen,
            })}
          />
        </ButtonBase>
      )}
      {!link && !item.subnav && (
        <Link
          to={
            home
              ? '/'
              : `/content/${item.title.toLowerCase().replace(/ /g, '-')}/`
          }
          onClick={() => setMenu(false)}
          className={clsx(
            classes.button,
            'zep-typo--normal-body1',
            'zep-button--full'
          )}
        >
          {item.title}
        </Link>
      )}
      {item.subnav && isButtonOpen && (
        <ul data-testid="subnav-list" className={classes.list}>
          {Object.keys(item.subnav).map((elem, idx) => (
            <li key={`sublistitem${idx}`} className={classes.listItem}>
              <Link
                to={`/content/${item.title
                  .toLowerCase()
                  .replace(/ /g, '-')}/${item.subnav[elem].title
                  .toLowerCase()
                  .replace(/ /g, '-')}/`}
                onClick={() => setMenu(false)}
                activeStyle={{ color: theme.colors.gray.gray.hex }}
                className={clsx(
                  classes.subnav,
                  'zep-typo--normal-body1',
                  'zep-button--full'
                )}
                data-testid="sidebarNavItemLink"
              >
                {item.subnav[elem].title}
              </Link>
            </li>
          ))}
        </ul>
      )}
    </li>
  )
}

SidebarNavItem.propTypes = {
  classes: PropTypes.object.isRequired,
  item: PropTypes.object.isRequired,
  setMenu: PropTypes.func.isRequired,
}

export default withStyles(styles)(SidebarNavItem)
