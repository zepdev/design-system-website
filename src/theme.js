import { createMuiTheme } from '@material-ui/core/styles'
import guidelines from '@zlab-de/zel-react/guidelines.json'

const theme = createMuiTheme({
  palette: {
    type: 'light',
    primary: {
      light: guidelines.theme.indigo.secondary,
      main: guidelines.theme.indigo.primary,
      dark: guidelines.theme.indigo.primary,
    },
    error: { main: guidelines.color.support.alert },
  },
  status: {
    gray: '#eceeef',
  },
  logo: guidelines.logo,
  blue: guidelines.theme.blue,
  indigo: guidelines.theme.indigo,
  red: guidelines.theme.red,
  yellow: guidelines.theme.yellow,
  guidelines: guidelines,
  space: guidelines.spacing.component,
  color: guidelines.color,
})

export default theme
