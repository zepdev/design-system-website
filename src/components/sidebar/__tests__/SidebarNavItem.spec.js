import React from 'react'
import { render, fireEvent } from '@testing-library/react'
import { ThemeProvider } from 'react-jss'
import theme from 'zeppelin-element-library/guidelines.json'
import SidebarNavItem from '../SidebarNavItem'

describe('SidebarNavItem', () => {
  it('renders correctly', () => {
    const mockProps = {
      setMenu: jest.fn(),
      item: {
        title: 'sectionTitle',
        subnav: {
          subSection: {
            title: 'subSectionTitle',
          },
        },
      },
    }
    const component = render(
      <ThemeProvider theme={theme}>
        <SidebarNavItem {...mockProps} />
      </ThemeProvider>
    )
    expect(component).toMatchSnapshot()
  })

  it('does not render sub-nav section when there is no sub-nav', () => {
    const mockProps = {
      setMenu: jest.fn(),
      item: {
        title: 'sectionTitle',
      },
    }
    const { queryByTestId, getByText } = render(
      <ThemeProvider theme={theme}>
        <SidebarNavItem {...mockProps} />
      </ThemeProvider>
    )
    // set isButtonOpen to equal true so that the section will render
    const button = getByText(mockProps.item.title)
    fireEvent.click(button)
    expect(queryByTestId('subnav-list')).toBeNull()
  })

  it('renders sub-nav when button is clicked', () => {
    const mockProps = {
      setMenu: jest.fn(),
      item: {
        title: 'sectionTitle',
        subnav: {
          subSection: {
            title: 'subSectionTitle',
          },
        },
      },
    }
    const { getByText, getByTestId } = render(
      <ThemeProvider theme={theme}>
        <SidebarNavItem {...mockProps} />
      </ThemeProvider>
    )
    const button = getByText(mockProps.item.title)
    fireEvent.click(button)
    expect(getByTestId('sidebarNavItemLink')).toHaveTextContent(
      mockProps.item.subnav.subSection.title
    )
  })
})
