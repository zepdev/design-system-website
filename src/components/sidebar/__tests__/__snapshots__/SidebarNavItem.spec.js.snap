// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SidebarNavItem renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <li
        class=""
      >
        <button
          class="SidebarNavItem-button-0-1-1 zep-typo--normal-body1 zep-button--full ButtonBase-button-0-1-9"
        >
          sectionTitle
          <svg
            aria-labelledby="title_chevronDown"
            class="SidebarNavItem-icon-0-1-5 SidebarNavItem-iconReset-0-1-7"
            height="20"
            viewBox="0 0 32 32"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <title
              id="title_chevronDown"
            >
              Chevron Down
            </title>
            <polygon
              fill="currentColor"
              points="7.41 8.58 12 13.17 16.59 8.58 18 10 12 16 6 10"
            />
          </svg>
        </button>
      </li>
    </div>
  </body>,
  "container": <div>
    <li
      class=""
    >
      <button
        class="SidebarNavItem-button-0-1-1 zep-typo--normal-body1 zep-button--full ButtonBase-button-0-1-9"
      >
        sectionTitle
        <svg
          aria-labelledby="title_chevronDown"
          class="SidebarNavItem-icon-0-1-5 SidebarNavItem-iconReset-0-1-7"
          height="20"
          viewBox="0 0 32 32"
          width="20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <title
            id="title_chevronDown"
          >
            Chevron Down
          </title>
          <polygon
            fill="currentColor"
            points="7.41 8.58 12 13.17 16.59 8.58 18 10 12 16 6 10"
          />
        </svg>
      </button>
    </li>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
